### PR TITLE
Methods cleanup and new DTLS "either" side methods

### PR DESCRIPTION
**Pull request by dgarske**
* Adds build option `WOLFSSL_EITHER_SIDE` for deferring the "side" of the TLS session until first connect or accept.
* Moved the TLS/DTLS method constructors to same place in tls.c.
* Added the DTLS generic v1.0 and v1.2 methods for "either" side.
* Added "either" methods unit tests.
* Added "either" `-v e` support to example client/server.
* Fix to expose `wolfSSL_use_certificate_file` and `wolfSSL_use_PrivateKey_file` without OPENSSL_EXTRA.
* Cleanup of the methods for (void)heap and log messages.

### CODE REVIEW

**Comment by dgarske** [comment]
Report of bug with this code using PK callbacks and DTLS where the code to allow missing private key isn't working with "either" case:

```
WOLFSSL accept error: server cert required
WOLFSSL wolfSSL error occurred, error = -317
```

Assigning back to me to fix.

**Comment by dgarske** [comment]
@toddouska : The issue turned out to be a coding bug on their side with not loading the server/client certificate using the WOLFSSL object and `wolfSSL_use_certificate_chain_buffer_format`. This PR is ready for peer review.

File: src/ssl.c [hunk 1]
```diff
@@ -9211,7 +9228,13 @@ int wolfSSL_DTLS_SetCookieSecret(WOLFSSL* ssl,
         if (ssl == NULL)
             return WOLFSSL_FATAL_ERROR;
 
-#ifdef WOLFSSL_NO_TLS12
+    #if defined(OPENSSL_EXTRA) || defined(WOLFSSL_EITHER_SIDE)
+        if (ssl->options.side == WOLFSSL_NEITHER_END) {
+            InitSSL_Side(ssl, WOLFSSL_SERVER_END);
```
Comment (toddouska) [inline src/ssl.c]: These Init functions HAVE to have their return codes checked.

File: src/ssl.c [hunk 2]
```diff
@@ -8849,35 +8878,38 @@ int wolfSSL_DTLS_SetCookieSecret(WOLFSSL* ssl,
         if (ssl == NULL)
             return BAD_FUNC_ARG;
 
-    #ifdef OPENSSL_EXTRA
+    #if defined(OPENSSL_EXTRA) || defined(WOLFSSL_EITHER_SIDE)
         if (ssl->options.side == WOLFSSL_NEITHER_END) {
-            ssl->options.side = WOLFSSL_CLIENT_END;
+            InitSSL_Side(ssl, WOLFSSL_CLIENT_END);
```
Comment (toddouska) [inline src/ssl.c]: These Init functions HAVE to have their return codes checked.

**Pull request merged**